packages/calypso-e2e/src/lib/components/sidebar-component.ts

```diff
@@ -35,32 +40,103 @@ export class SidebarComponent extends BaseContainer {
 	 */
 	async _postInit(): Promise< void > {
 		await this.page.waitForLoadState( 'domcontentloaded' );
+		this.sidebar = await this.page.waitForSelector( selectors.sidebar );
 	}
 
 	/**
-	 * Clicks on the sidebar menu item matching the name.
-	 * Note that the menu item must be visible in some shape or form.
-	 * If there are multiple elements that match the selector, the first
-	 * matching element will be clicked (as per Playwright documentation).
-	 * Waits and assertions should be placed on the page/component that this
-	 * method will cause a navigation to, to ensure readiness before the script
-	 * continues.
+	 * Given heading and subheading, or any combination of the two, locate and click on the items on the sidebar.
+	 *
+	 * This method supports any of the following use cases:
+	 *   - heading only
+	 *   - subheading only
+	 *   - heading and subheading
+	 *
+	 * Heading is defined as the top-level menu item that is permanently visible on the sidebar, unless outside
+	 * of the viewport.
 	 *
-	 * @param {string} name Plaintext name of the menu item in the sidebar.
+	 * Subheading is defined as the child-level menu item that is exposed only on hover or by toggling open the listing by clicking on the parent menu item.
+	 *
+	 * Note, in the current Nav Unification paradigm, clicking on certain combinations of sidebar menu items will trigger
+	 * navigation away to an entirely new page (eg. wp-admin). Attempting to reuse the SidebarComponent object
+	 * under this condition will throw an exception from the Playwright engine.
+	 *
+	 * @param {{[key: string]: string}} param0 Named object parameter.
+	 * @param {string} param0.heading Plaintext representation of the top level heading.
+	 * @param {string} param0.subheading Plaintext representation of the child level heading.
 	 * @returns {Promise<void>} No return value.
 	 */
-	async clickMenuItem( name: string ): Promise< void > {
-		await this.page.click( `text=${ toTitleCase( name ) }` );
+	async gotoMenu( {
+		heading,
+		subheading,
+	}: {
+		heading?: string;
+		subheading?: string;
+	} ): Promise< void > {
+		if ( heading ) {
+			heading = toTitleCase( heading ).trim();
+			// This will exclude entries where the `heading` term matches multiple times
+			// eg. `Settings` but they are sub-headings in reality, such as Jetpack > Settings.
+			// Since the sub-headings are always hidden unless heading is selected, this works to
+			// our advantage.
+			const selector = `${ selectors.heading } span:has-text("${ heading }"):visible`;
+			await Promise.all( [ this.page.waitForNavigation(), this._click( { selector } ) ] );
```

Comment:
Could you clarify a bit more? Which heading/item behaves this way?

Reply:
Sure! In the mobile viewport, some items are direct links that require waiting for navigation, and some are just toggles for opening the submenu. This is not the case for the desktop viewport where clicking anything will immediately redirect.

Reply:
I see. I can try a selection of the heading/items with the mobile viewport to check how this code works.

Reply:
Well you see, I didn't even take mobile viewport into account since there's nothing that opens the sidebar into view in the first place.

I'll revert this `Promise.all` for the time being, and I will file a new task to get mobile support going.

Reply:
Filed #53803 for adding mobile support.